### PR TITLE
RSDK-767 - add esp32 i2c implementation

### PR DESCRIPTION
**Pull request by gvaradarajan**

### CODE REVIEW

File: src/esp32/board.rs [hunk 1]
```diff
@@ -37,20 +38,26 @@ pub(crate) fn register_models(registry: &mut ComponentRegistry) {
...
     pub fn new(
         pins: Vec<PinDriver<'static, AnyOutputPin, Output>>,
         analogs: Vec<Rc<RefCell<dyn AnalogReader<u16, Error = anyhow::Error>>>>,
     ) -> Self {
-        EspBoard { pins, analogs }
+        let i2cs = HashMap::new();
+        EspBoard {
+            pins,
+            analogs,
+            i2cs,
```
Comment (npmenard) [inline src/esp32/board.rs:53]: nit: prefer i2c : HashMap::new()

File: src/esp32/i2c.rs [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    pub bus: &'static str,
+    pub baudrate_hz: u32,
+    pub timeout_ns: u32,
+    pub data_pin: i32,
+    pub clock_pin: i32,
+}
+
+impl From<Esp32I2cConfig> for I2cConfig {
+    fn from(value: Esp32I2cConfig) -> I2cConfig {
+        // TODO: when next version of esp_idf_hal is released, use below instead
```
Comment (npmenard) [inline src/esp32/i2c.rs:23]: is it covered in the bumped version we just merged?

File: src/esp32/i2c.rs [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+                    driver,
+                    timeout_ns,
+                })
+            }
+            _ => anyhow::bail!("only i2c0 or i2c1 supported, i2c bus must match either value"),
+        }
+    }
+}
+
+impl<'a> I2CHandle<u8> for Esp32I2C<'a> {
```
Comment (npmenard) [inline src/esp32/i2c.rs:157]: Are you planning to remove the template on I2CHandle for the address type?

**Review by npmenard — APPROVED** [review]
LGTM can't wait to see it in action

File: src/esp32/i2c.rs (hunk 3 above)
Comment (gvaradarajan) [reply src/esp32/i2c.rs:157]: oh, I forgot I accommodated that. I'll remove it, but out of curiosity do you know of any processors that use 10-bit addresses for I2C?

Comment (npmenard) [reply src/esp32/i2c.rs:157]: not to mu knowledge

File: src/esp32/i2c.rs (hunk 2 above)
Comment (gvaradarajan) [reply src/esp32/i2c.rs:22]: No the version we bumped to (the latest) was released in December and the change allowing timeout configuration was committed in February

**Review by npmenard — APPROVED** [review]
LGTM

**Pull request merged**